### PR TITLE
remove dependency on facets

### PR DESCRIPTION
**Pull request by JackVCurtis**

### CODE REVIEW

File: server/models/ParticipantsModel/index.js [hunk 1]
```diff
@@ -18,44 +18,31 @@ const ParticipantsModel = {
...
-        },
-      }
-
-      return await db
-        .collection(collections.assessmentDayData)
-        .find(chartQuery, { projection: ALL_SUBJECTS_MONGO_PROJECTION })
-        .stream()
+      .collection(collections.assessmentDayData)
+      .find(query, { projection: ALL_SUBJECTS_MONGO_PROJECTION })
+      .stream()
```
Comment (mikestone14) [inline server/models/ParticipantsModel/index.js]: nit but this indentation looks off. Surprised prettier didn't catch it.

File: server/services/FiltersService/index.js [hunk 2]
```diff
@@ -66,149 +66,40 @@ class FiltersService {
...
-      ]
+              {assessment: INCLUSION_EXCLUSION_CRITERIA_FORM},
+              {'dayData.included_excluded': {$in: includedValues } }
+            ]
+          })
+          break
+        case 'chrcrit_part':
+          filterQueries.push({
+            $and: [
+              {assessment: INCLUSION_EXCLUSION_CRITERIA_FORM},
```
Comment (mikestone14) [inline server/services/FiltersService/index.js]: another nit on formatting. Can you run prettier?

**Review by mikestone14 — COMMENTED** [review]
overall this looks good to me but I'll someone with more mongo expertise approve 😄

**Comment by Elkrival** [comment]
@JackVCurtis 


With the new changes the count is the following
Screenshot 2024-03-01 at 9 45 19 AM

With the original code the result is different 
Screenshot 2024-03-01 at 9 44 38 AM

The data is from the dataset https://www.dropbox.com/scl/fi/whzexi3resv8tgpie6bsj/data_for_gnar_anon.tar.gz?rlkey=xm00gkbcsp6bhol17ug3pcu1y&dl=0 and using the subject counts chart from yesterday

File: server/services/FiltersService/index.js [hunk 3]
```diff
@@ -66,149 +66,40 @@ class FiltersService {
...
-          },
-        },
-        {
-          $addFields: {
-            filter: INCLUSION_EXCLUSION_CRITERIA_FORM,
-          },
-        },
-      ]
+              {assessment: INCLUSION_EXCLUSION_CRITERIA_FORM},
+              {'dayData.included_excluded': {$in: includedValues } }
```
Comment (Elkrival) [inline server/services/FiltersService/index.js]: will this query work without using  $elemMatch..? 
Same for the other ones below

**Review by Elkrival — COMMENTED** [review]
I like that it removes a lot of code, there's some polishing to do with the queries to match counts

File: server/services/FiltersService/index.js (hunk 3 above)
Comment (JackVCurtis) [reply server/services/FiltersService/index.js]: I don't understand what you mean, it is working without using $elemMatch. Are you asking me whether the query works as written?

File: server/services/FiltersService/index.js (hunk 2 above)
Comment (JackVCurtis) [reply server/services/FiltersService/index.js]: We don't have prettier installed or configured for this app.

File: server/models/ParticipantsModel/index.js (hunk 1 above)
Comment (JackVCurtis) [reply server/models/ParticipantsModel/index.js]: [formatting fixes](https://github.com/AMP-SCZ/dpdash/pull/659/commits/ad8bf5c42ecf247244eda6a06081123d225c4863)

File: server/services/FiltersService/index.js (hunk 2 above)
Comment (JackVCurtis) [reply server/services/FiltersService/index.js]: [formatting fixes](https://github.com/AMP-SCZ/dpdash/pull/659/commits/ad8bf5c42ecf247244eda6a06081123d225c4863)

**Comment by JackVCurtis** [comment]
> @JackVCurtis
> 
> With the new changes below the count is the following Screenshot 2024-03-01 at 9 45 19 AM
> 
> With the original code the result is different Screenshot 2024-03-01 at 9 44 38 AM
> 
> The data is from the dataset https://www.dropbox.com/scl/fi/whzexi3resv8tgpie6bsj/data_for_gnar_anon.tar.gz?rlkey=xm00gkbcsp6bhol17ug3pcu1y&dl=0 and using the subject counts chart from yesterday

How do we tell which count is correct? I don't trust the previous code to be more accurate than the current code, frankly.

**Comment by Elkrival** [comment]
@JackVCurtis the way we know the counts are correct is because the count logic has been vetted and tested by the client

File: server/services/FiltersService/index.js (hunk 3 above)
Comment (Elkrival) [reply server/services/FiltersService/index.js]: Just curious about this because I like the way it looks written without using an index or elemMatch

**Comment by Elkrival** [comment]
@JackVCurtis playing more with it, I learned that the initial filters aren't kicking in, the result of 1301 is the same result as if I had requested a chart without any filters

File: server/services/FiltersService/index.js [hunk 4]
```diff
@@ -66,149 +66,40 @@ class FiltersService {
...
-      $facet.formInclusion = [
-        {
-          $match: {
-            assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+  get filterQueries() {
+    const filterQueries = []
+    for (let filterKey of Object.keys(this.filters)) {
+      const filter = this.filters[filterKey]
+      const includedFilters = filter.filter(({_, value}) => value === TRUE_STRING)
+      const includedValues = includedFilters.map(({name, _}) => FILTER_TO_MONGO_VALUE_MAP[name])
```
Comment (Elkrival) [inline server/services/FiltersService/index.js]: Can we rename this variable..?
```suggestion
      const includedValues = includedFilters.map(({name, _}) => FILTER_TO_MONGO_VALUE_MAP[name])
```

File: server/services/FiltersService/index.js [hunk 5]
```diff
@@ -66,149 +66,40 @@ class FiltersService {
...
-    if (isIncAndCrit) {
-      $facet.formInclusion = [
-        {
-          $match: {
-            assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+  get filterQueries() {
+    const filterQueries = []
+    for (let filterKey of Object.keys(this.filters)) {
+      const filter = this.filters[filterKey]
+      const includedFilters = filter.filter(({_, value}) => value === TRUE_STRING)
```
Comment (Elkrival) [inline server/services/FiltersService/index.js]:
```suggestion
      const includedFilters = filter.filter(({ value}) => value === TRUE_STRING)
```

File: server/services/FiltersService/index.js (hunk 4 above)
Comment (Elkrival) [inline server/services/FiltersService/index.js]:
```suggestion
      const includedValues = includedFilters.map(({name}) => FILTER_TO_MONGO_VALUE_MAP[name])
```

File: server/models/ParticipantsModel/index.js [hunk 6]
```diff
@@ -29,33 +29,20 @@ const ParticipantsModel = {
...
+      .distinct('participant', {
+        $or: filtersService.filterQueries
+      })
+    return await db
+      .collection(collections.assessmentDayData)
+      .aggregate([
+        { $match: { study: { $in: filters.sites, $nin: STUDIES_TO_OMIT }} },
+        { $match: { participant: { $in: includedParticipants }}},
+        { $match: { assessment: chart.assessment }}
+      ])
```
Comment (Elkrival) [inline server/models/ParticipantsModel/index.js]: It's non-blocking suggestion.
```suggestion
    return await db
      .collection(collections.assessmentDayData)
      .aggregate([
        { $match: { assessment: chart.assessment ,study: { $in: filters.sites, $nin: STUDIES_TO_OMIT }} },
        { $match: { participant: { $in: includedParticipants }}},
       ])
```

**Review by Elkrival — COMMENTED** [review]
small things

File: server/services/FiltersService/index.js (hunk 3 above)
Comment (Elkrival) [reply server/services/FiltersService/index.js]: did some more reading on this, on querying an array of embedded documents and this syntax is great 😄

File: server/services/FiltersService/index.js [hunk 7]
```diff
@@ -66,149 +76,41 @@ class FiltersService {
...
+    const filterQueries = []
+    const filterNames = new Set(Object.keys(this.filters).filter(key => (this.filters[key].map(f => f.value).indexOf(TRUE_STRING) > -1)))
+
+    if (filterNames.has(INCLUSION_EXCLUSION_KEY) && filterNames.has(CHRCRIT_KEY)) {
+      filterQueries.push({
+        assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+        ...includedExcludedQuery(filterToMongoValues(this.filters[INCLUSION_EXCLUSION_KEY])),
+        ...chrChritQuery(filterToMongoValues(this.filters[CHRCRIT_KEY]))
+      })
+    } else if (filterNames.has(INCLUSION_EXCLUSION_KEY)) {
```
Comment (mikestone14) [inline server/services/FiltersService/index.js:94]: just checking that these should all be `else if`s so only one is applied vs `if`s where multiple could be applied.

**Comment by Elkrival** [comment]
@JackVCurtis filters are being applied after the fact, I've posted a ticket for a small bug. Otherwise g2g

**Review by Elkrival — APPROVED** [review]
Couple of tests need to be fixed

**Pull request merged**